memory/function/kaslr/runtest.sh

```diff
@@ -0,0 +1,315 @@
+		get_symbol_addr_snapshot
+		for f in $stable_file_list; do
+			rlAssertEquals "$f should be default" "${!f}" "$(cat $f)"
+		done
+	fi
+
+	local i=0
+	if [ "$current_state" = "after_r_nokaslr_compare" ]; then
+		rlReport "reboot" PASS
+		rlAssertGrep nokaslr /proc/cmdline
+		((i++))
+		for f in $cmp_file_list; do
+			rlAssertEquals "$f should be same" "$(cat ${f}.old)" "$(cat $f)"
+		done
+		slub_freelist_random 0 $i
+		rlRun "grubby --remove-args nokaslr --update-kernel ALL"
+		rlPhaseEnd
+		rhts-reboot
+	elif [ "$current_state" = "after_r_nokaslr_cleanup" ]; then
+		rlReport "reboot" PASS
+		rlAssertNotGrep nokaslr /proc/cmdline
+	elif [ "$current_state" = "before_r_nokaslr_snapshot" ]; then
+		rlAssertGrep nokaslr /proc/cmdline
+		rlPhaseEnd
+		rhts-reboot
+	else
+		rlAssertGrep nokaslr /proc/cmdline
+		slub_freelist_random 0 $i
+		rlPhaseEnd
+		rhts-reboot
+	fi
+}
+
+function run_kaslr_x8664()
+{
+	grep CONFIG_RANDOMIZE_MEMORY=y /boot/config-"$(uname -r)" 2>/dev/null || { rlReport "Skip-not-support" PASS; return; }
+	get_kernel_version
+	if  [ "$kver_major" -lt 3 ]; then
+		rlReport "Skip-not-support" PASS
+		return
+	elif [ "$kver_major" -eq 3 ] && [ "$kver_minor" -eq 10 ]; then
+		if [ "$krel_major" -lt 705 ]; then
+			rlReport "Skip-not-support" PASS
+			return
+		fi
+	fi
+	rlPhaseStartTest $phase
+		rlRun "sed -i '1d' TEST_STATE" 0 "to next state $(sed -n '2p' TEST_STATE)"
+		case "$phase" in
+		kaslr*)
+			x86_kaslr_test
+			;;
+		nokaslr*)
+			x86_nokaslr_test
+			;;
+		esac
+
+	rlPhaseEnd
+}
+
+function prepare_state_q()
+{
+	touch TEST_STATE
+	if grep nokaslr /proc/cmdline; then
+		echo "default cmdline: $(cat /proc/cmdline)"
+		echo "before_r_nokaslr_snapshot" > TEST_STATE
+		echo "after_r_nokaslr_compare" >> TEST_STATE
+		echo "after_r_kaslr_snapshot" >> TEST_STATE
+		echo "after_r_kaslr_compare" >> TEST_STATE
+		echo "after_r_kaslr_cleanup" >> TEST_STATE
+	else
+		echo "before_r_kaslr_snapshot" > TEST_STATE
+		echo "after_r_kaslr_compare" >> TEST_STATE
+		echo "after_r_nokaslr_snapshot" >> TEST_STATE
+		echo "after_r_nokaslr_compare" >> TEST_STATE
+		echo "after_r_nokaslr_cleanup" >> TEST_STATE
+	fi
+	rlRun -l "cat TEST_STATE" 0 "states we are goting to handle."
+}
+
+function get_kernel_version()
+{
+    kver_major=$(uname -r | cut -d- -f1 | cut -d. -f 1) # 3
+    kver_minor=$(uname -r | cut -d- -f1 | cut -d. -f 2) # 10
+    kver_mminor=$(uname -r | cut -d- -f1 | cut -d. -f 3) # 0
+    krel_major=$(uname -r | cut -d- -f2 | cut -d. -f 1) # 514
+}
+
+rlJournalStart
+	if ! test -f SETUP_FINISH; then
+		rlPhaseStartSetup
+			prepare_state_q
+			grep nokaslr /proc/cmdline && rlLogInfo "nokaslr in cmdline: $(cat /proc/cmdline)"
+			unset ARCH
+			if [ "$SLUB_RANDOM" -ne 0 ]; then
+				uname -r | grep debug && { rpm -q kernel-devel-debug || yum -y install kernel-devel-debug; }
+				# slub freelist random testing
+				yum -y install libelf-dev || yum -y install libelf-devel || yum -y install elfutils-libelf-devel
```

Comment:
The tests we've on-boarded use a yum/dnf selector like his one https://github.com/CKI-project/tests-beaker/blob/master/distribution/kpkginstall/runtest.sh#L161 Would you mind updating it to use something similar ?

Reply:
Sure, I updated the commit.